### PR TITLE
Xamarin support

### PR DESCRIPTION
**Pull request by odra**
**Describe what this PR does and why we need it**:

Adds support to xamarin client app type so a xamarin-apb repository can be created in aerogearcatalog.

Changes proposed in this pull request
 - Adds xamarin support in mobile cli

**Does this PR depend on another PR (Use this to track when PRs should be merged)**
depends-on <PR>

It depends on a xamarin APB pr which was not created yet.

**Which issue this PR fixes (This will close that issue when PR gets merged)**
https://issues.jboss.org/browse/AEROGEAR-2405

### CODE REVIEW

**Comment by witmicko** [comment]
Duplicate of https://github.com/aerogear/mobile-cli/pull/96

File: pkg/cmd/clients.go [hunk 1]
```diff
@@ -126,12 +126,12 @@ Run the "mobile get clients" command from this tool to get the client ID.`,
...
 func (cc *ClientCmd) CreateClientCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "client <name> <clientType iOS|cordova|android> <appIdentifier bundleID|packageName>",
+		Use:   "client <name> <clientType iOS|cordova|android|xamarin> <appIdentifier bundleID|packageName>",
 		Short: "create a mobile client representation in your namespace",
 		Long: `create client sets up the representation of a mobile application of the specified type in your namespace.
 This is used to provide a mobile client context for various actions such as creating, starting or stopping mobile client builds.
 
-The available client types are android, cordova and iOS. 
+The available client types are android, cordova and iOS, xamarin.
```
Comment (philbrookes) [inline pkg/cmd/clients.go]: This should read:
```
The available client types are android, cordova, iOS and xamarin. 
```

File: pkg/cmd/clients_test.go [hunk 2]
```diff
@@ -417,11 +417,48 @@ func TestMobileClientsCmd_TestCreateClient(t *testing.T) {
...
+					t.Fatal("expected the client name to be test but got ", c.Spec.Name)
+				}
+				if c.Spec.ApiKey == "" {
+					t.Fatal("expected an apiKey to be generated but it was empty")
+				}
+				icon, ok := c.Annotations["icon"]
+				if !ok {
+					t.Fatal("expected an icon to be set but there was none")
+				}
+				if icon != "fa fa-xamarin" {
```
Comment (philbrookes) [inline pkg/cmd/clients_test.go]: is this passing? It looks like it is a different value [here](https://github.com/aerogearcatalog/xamarin-app-apb/pull/1/files#diff-bd84d488df2d1137568929b31a1f9c71R10)?

File: pkg/cmd/clients.go [hunk 3]
```diff
@@ -172,6 +172,11 @@ When used standalone, a namespace must be specified by providing the --namespace
 			case "cordova":
 				app.Annotations["icon"] = "font-icon icon-cordova"
 				break
+			case "xamarin":
+				app.Annotations["icon"] = "icon icon-xamarin"
```
Comment (sedroche) [inline pkg/cmd/clients.go]: Should this to be the same as cordova `font-icon`?

**Pull request merged**